src/htm/algorithms/Connections.hpp

```diff
@@ -434,6 +476,18 @@ class Connections : public Serializable
   void raisePermanencesToThreshold(const Segment    segment,
                                    const UInt       segmentThreshold);
 
+
+  /**
+   *  iteration: ever increasing step count. 
+   *  Increases each main call to "compute". Since connections has more
+   *  methods that are called instead of compute (adaptSegment, computeActivity,..)
+   *  this counter is increased in @ref `computeActivity` as it is called by both
+   *  SP & TM. 
+   */
+//!  const UInt32& iteration = iteration_; //FIXME cannot construct iteration like this?
+  UInt32 iteration() const { return iteration_; }
```

Comment:
know how could I write this to use the const ref?

Reply:
@ctrl-z-9000-times do you know why the commented above does not work?

Reply:
Not exactly, but C++ is really picky about the `const &` trick.  It stops the compiler from auto-generating certain boilerplate methods.  I think I've encountered stuff like this before, but I don't remember how I fixed it.